M2/Macaulay2/packages/Msolve.m2

```diff
@@ -0,0 +1,580 @@
+newPackage(
+	"Msolve",
+	Version => "1.23", 
+    	Date => "March 2024",
+    	Authors => {{Name => "Martin Helmer", 
+		  Email => "mhelmer@ncsu.edu", 
+		  HomePage => "http://martin-helmer.com/"}, {Name => "Mike Stillman", 
+		  Email => "mike@math.cornell.edu", 
+		  HomePage => "https://math.cornell.edu/michael-e-stillman"},{Name => "Anton Leykin", 
+		  Email => "leykin@math.gatech.edu", 
+		  HomePage => "https://antonleykin.math.gatech.edu/"}},
+    	Headline => "An interface to the msolve package (https://msolve.lip6.fr/) which computes Groebner Basis and does real root isolation",
+    	AuxiliaryFiles => true,
+	DebuggingMode => false,       
+	Configuration => {"msolveExecutable"=>"msolve"}
```

Comment:
Would it make sense to use [`findProgram`](https://macaulay2.com/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_find__Program.html) and [`runProgram`](https://macaulay2.com/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_run__Program.html) for this package?  Then, for example, we could set the path to `msolve` using [`programPaths`](https://macaulay2.com/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_program__Paths.html) instead of adding this configuration option.

Reply:
Also, should there be an `OptionalComponentsPresent` option since msolve may or may not be present on the system?  (And should we install it on the github builds so this package is tested?)

Reply:
It's already in Ubuntu and brew I believe.

Reply:
It's been in Ubuntu since 23.10, and it's also available in the Macaulay2 PPA for 18.04, 20.04, and 22.04.

Reply:
We have added `OptionalComponentsPresent` but not sure whether `runProgram` is useful at this point in this case.

We don't seem to know a way to get `msolve` to output its version 🤪

Reply:
`findProgram`/`runProgram` ire useful, I think, because if msolve isn't already on the system and ends up getting built and installed in `currentLayout#"programs"` (which isn't placed on the `PATH` by default), then we'll still be able to find it.